internal/provider/provider.go

```diff
@@ -147,7 +148,13 @@ func (p *btpcliProvider) Configure(ctx context.Context, req provider.ConfigureRe
 	}
 
 	client := btpcli.NewClientFacade(btpcli.NewV2ClientWithHttpClient(p.httpClient, u))
-	client.UserAgent = fmt.Sprintf("Terraform/%s terraform-provider-btp/%s", req.TerraformVersion, version.ProviderVersion)
+	btp_append_user_agent := os.Getenv("BTP_APPEND_USER_AGENT")
+
+	if len(strings.TrimSpace(btp_append_user_agent)) == 0 {
+		client.UserAgent = fmt.Sprintf("Terraform/%s terraform-provider-btp/%s", req.TerraformVersion, version.ProviderVersion)
+	} else {
+		client.UserAgent = fmt.Sprintf("Terraform/%s terraform-provider-btp/%s modified-user-agent/%s", req.TerraformVersion, version.ProviderVersion, btp_append_user_agent)
```

Comment:
I would rename that to `custom-user-agent` as this is not a modified version but an additional information